src/ResourceManager/SubscriptionDefinition/Commands.SubscriptionDefinition/Cmdlets/NewAzureRmSubscriptionDefinition.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.SubscriptionDefinition.Common;
+using Microsoft.Azure.Commands.SubscriptionDefinition.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+
+namespace Microsoft.Azure.Commands.SubscriptionDefinition.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmSubscriptionDefinition", SupportsShouldProcess = true), OutputType(typeof(PSSubscriptionDefinition))]
+    public class NewAzureRmSubscriptionDefinition : AzureSubscriptionDefinitionCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = "Name of the subscription definition.")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Offer type of the subscription definition.")]
+        [ArgumentCompleter(typeof(OfferTypeCompleter))]
+        public string OfferType { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Display name of the subscription.")]
+        public string SubscriptionDisplayName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.ShouldProcess(target: this.Name, action: "Create subscription definition"))
+            {
+                this.WriteSubscriptionDefinitionObject(this.SubscriptionDefinitionsClient.SubscriptionDefinitions.Create(this.Name, new Microsoft.Azure.Management.ResourceManager.Models.SubscriptionDefinition(
+                    offerType: this.OfferType,
+                    subscriptionDisplayName: this.SubscriptionDisplayName ?? this.Name)));
+            }
+        }
+
+        private class OfferTypeCompleter : IArgumentCompleter
```

Comment:
@wilcobmsft this is awesome! We will need to tweak this a little since `IArgumentCompleter` isn't support in PowerShell 3.0 or 4.0 (which we currently support, but hopefully not much longer) and move it into a PowerShell script that gets executed when the module is loaded.

@maddieclayton would you mind helping Wilco move this functionality to the proper location?

Reply:
Maddie's working on a general purpose, backward compatible argument completer helper. Once that's available, I'll switch to that. For now, I'll just remove the argument completer to avoid breaking PS 3.0.